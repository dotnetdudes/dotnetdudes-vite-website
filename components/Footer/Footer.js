// <Footer></Footer>
import FooterTemplate from './template';

class Footer extends HTMLElement {
  constructor() {
    super();
  }
  connectedCallback() {
    this.innerHTML = FooterTemplate;
  }
  disconnectedCallback() {
    //
  }
  attributeChangedCallback(attrName, oldVal, newVal) {
    //
  }
}

window.customElements.define('page-footer', Footer);