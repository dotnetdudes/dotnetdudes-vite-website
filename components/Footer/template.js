const FooterTemplate = /*html*/ `
<footer>
    <div>
        <div class="copyright" itemprop="copyrightNotice">
            &copy; Copyright <strong>Dotnetdudes</strong>. All rights
            reserved.
            <div
                itemprop="publisher"
                itemscope
                itemtype="https://schema.org/Organization"
            >
                <div class="copyright-name">Designed and developed by <span itemprop="name">Dotnetdudes</span></div>
            </div>
        </div>
       
        <div class="credits">UIcons by <a href="https://www.flaticon.com/uicons">Flaticon</a></div>
    </div>
</footer>
`;

export default FooterTemplate;
