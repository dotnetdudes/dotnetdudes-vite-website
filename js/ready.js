// import { getInitialTheme, setTheme } from './theme';

export const onSiteLoaded = () => {
    try {

        document.addEventListener('DOMContentLoaded', function () {
            // const theme = getInitialTheme();
            // console.log(theme);
            // setTheme(theme);
            // const pnl = document.getElementById('loadingpanel');
            // pnl.style.display = 'none';
            console.log('DOMContentLoaded');
        });
    } catch (error) {
        console.log(error);
    }
};
