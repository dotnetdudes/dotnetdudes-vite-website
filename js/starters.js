export const onStarterLoaded = () => {
    try {
        const openReactStarter = () => {
            window.open("https://github.com/dotnetdudes/react-starter", "_blank");
        }
        const openReactNativeStarter = () => {
            window.open("https://github.com/dotnetdudes/react-native-starter", "_blank");
        }
        document.addEventListener('DOMContentLoaded', function () {
            const btn = document.getElementById('reactStarter');
            const reactBtn = document.getElementById('nativeStarter');
            btn.addEventListener('click', openReactStarter);
            reactBtn.addEventListener('click', openReactNativeStarter);
        });
    } catch (error) {
        console.log(error);
    }
}
