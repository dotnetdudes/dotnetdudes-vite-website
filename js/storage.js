export const setItem = (name, item) => {
    if (!window.localStorage) return false;
    window.localStorage.setItem(name, item);
    return true;
};

export const getItem = (name) => {
    if (!window.localStorage) return false;
    return window.localStorage.getItem(name);
};

export const clearStorage = () => {
    if (!window.localStorage) return false;
    window.localStorage.clear();
};
