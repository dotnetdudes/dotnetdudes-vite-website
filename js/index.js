import { onSiteLoaded } from './ready';
/* import dotenv from 'dotenv';
const dotenv_result = dotenv.config({ silent: true }); */

// onSiteLoaded();
console.log('okay');
