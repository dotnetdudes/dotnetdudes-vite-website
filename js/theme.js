import { getItem, setItem } from "./storage";

export const isDarkMode = () =>
  window.matchMedia &&
  window.matchMedia("(prefers-color-scheme: dark)").matches;

export const getInitialTheme = () => {
  let initialTheme = isDarkMode() ? "dark" : "light";
  // console.log(isDarkMode());
  if (getItem("dudesTheme")) initialTheme = getItem("dudesTheme");
  return initialTheme;
};

export const setTheme = (theme) => {
  // set css link href
  setItem("dudesTheme", theme);
};
