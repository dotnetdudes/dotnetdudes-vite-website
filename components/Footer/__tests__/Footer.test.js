import { test, expect, describe } from "vitest";
import "../Footer.js";

describe("Footer", () => {
  test("Footer component should match current snapshot", () => {
    document.body.innerHTML = `<page-footer></page-footer>`;
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
