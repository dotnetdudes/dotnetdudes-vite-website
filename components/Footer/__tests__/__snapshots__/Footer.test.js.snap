// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Footer component should match current snapshot 1`] = `
"<page-footer>
<footer>
    <div>
        <div class="copyright" itemprop="copyrightNotice">
            Â© Copyright <strong>Dotnetdudes</strong>. All rights
            reserved.
            <div itemprop="publisher" itemscope="" itemtype="https://schema.org/Organization">
                <div class="copyright-name">Designed and developed by <span itemprop="name">Dotnetdudes</span></div>
            </div>
        </div>
       
        <div class="credits">UIcons by <a href="https://www.flaticon.com/uicons">Flaticon</a></div>
    </div>
</footer>
</page-footer>"
`;
